yugabyte/run-jepsen.py

```diff
@@ -55,68 +65,172 @@
     "partition-one",
     "partition-ring",
-    "partition",
-    # "clock-skew",
+    "partition"
 ]
 
 SCRIPT_DIR = os.path.abspath(os.path.dirname(sys.argv[0]))
 STORE_DIR = os.path.join(SCRIPT_DIR, "store")
+LOGS_DIR = os.path.join(SCRIPT_DIR, "logs")
 SORT_RESULTS_SH = os.path.join(SCRIPT_DIR, "sort-results.sh")
 
 child_processes = []
 
+
 def cleanup():
     deadline = time.time() + 5
     for p in child_processes:
-        while p.poll() == None and time.time() < deadline:
+        while p.poll() is None and time.time() < deadline:
             time.sleep(1)
         try:
             p.kill()
         except OSError, e:
             if e.errno != errno.ESRCH:
                 raise e
 
-def run_cmd(cmd, shell=True, timeout=None, exit_on_error=True):
-    print cmd
-    p = subprocess.Popen(cmd, shell=True)
-    child_processes.append(p)
 
-    if timeout:
-        deadline = time.time() + timeout
-    while p.poll() == None and (timeout == None or time.time() < deadline):
-        time.sleep(1)
+def run_cmd(cmd, shell=True, timeout=None, exit_on_error=True, log_name_prefix=None):
+    logging.info("Running command: %s", cmd)
+    if log_name_prefix is not None:
+        timestamp_str = time.strftime('%Y-%m-%dT%H-%M-%S')
+        log_name_prefix += '_' + timestamp_str
+        stdout_path = os.path.join(LOGS_DIR, log_name_prefix + '_stdout.log')
+        stderr_path = os.path.join(LOGS_DIR, log_name_prefix + '_stderr.log')
+        logging.info("stdout log: %s, stderr log: %s", stdout_path, stderr_path)
+
+    stdout_file = None
+    stderr_file = None
+
+    popen_kwargs = dict(shell=True)
+    try:
+        if log_name_prefix is None:
+            p = subprocess.Popen(cmd, **popen_kwargs)
+        else:
+            stdout_file = open(stdout_path, 'wb')
+            stderr_file = open(stderr_path, 'wb')
+            p = subprocess.Popen(cmd, stdout=stdout_file, stderr=stderr_file, **popen_kwargs)
+
+        child_processes.append(p)
+
+        if timeout:
+            deadline = time.time() + timeout
+        while p.poll() is None and (timeout is None or time.time() < deadline):
+            time.sleep(1)
 
-    if p.poll() == None:
-        timed_out = True
-        p.terminate()
+        if p.poll() is None:
+            timed_out = True
+            p.terminate()
+        else:
+            timed_out = False
+
+        child_processes.remove(p)
+        if exit_on_error and p.returncode != 0:
+            logging.error(
+                    "Failed running command (exit code: %d): %s",
+                    p.returncode, cmd)
+            sys.exit(p.returncode)
+        return CmdResult(returncode=p.returncode, timed_out=timed_out)
+    finally:
+        if stdout_file is not None:
+            stdout_file.close()
+        if stderr_file is not None:
+            stderr_file.close()
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        '--tarball-url',
```

Comment:
We can use just `url`, shorter arg name is usually easier to use and there is no ambiguity for now.